qtpy/__init__.py

```diff
@@ -150,7 +154,9 @@ class PythonQtWarning(Warning):
 
             del macos_version
     except ImportError:
-        API = os.environ['QT_API'] = 'pyqt6'
+        API = 'pyqt6'
```

Comment:
Not sure, I just tried not to change it in this PR.

From a quick look, the default is PyQt5 (i.e., it's what API is if QT_API is not set), and there currently is no differentiation between whether a user had set it or not at this point in the code. And the docs say the bindings will be tried in some specific order, and I think that order is meant to be followed in the code.

Given that this stuff is not trivial, and the fallback order is (or at least should be!) unchanged in this PR, I'd rather not tackle that here, and leave that for another PR, probably for someone else after some additional discussion. For example, if I just committed all the suggested changes about which API should come next, I don't think the code will work as desired -- it's currently framed around an API "try order" that gets followed, so below when you suggest changing the pyside2-fail-path (line 194) to set the `API = 'pyqt5'` instead of `API = 'pyside6'`, this will not have the intended (I think?) effect of actually *trying* pyqt5, because that happens earlier in the code (up here starting around line 130), whereas `'pyside6'` does currently get tried because it comes later in the code path... so I think more time/thought/effort/refactoring would be necessary to achieve the goal of the suggested changes.

Reply:
I think the change @ccordoba12 wanted to suggest is changing the _`try order`_ to be `PyQt5, PySide2, PyQt6, PySide6` (which will need not only to change the values on the `else` statements but also moving the whole blocks for the respective binding to follow such order). However, as @larsoner said I think that's kind of out the scope for this PR. The full behavior described (some sort of intelligent fallback) maybe its related with #154 (so if you try PyQt5 fallback to PySide2, but also if you try PySide2 fallback to PyQt5 and so on).

Just in case @ccordoba12 , I think the _`try order`_ on `v1.x` was something like `PyQt5, PySide2, PyQt4, PySide` but after the removal of Qt4 and addition of Qt6 the order on `v2.x/master` is now `PyQt5, PyQt6, PySide2, PySide6`. So maybe opening a new issue for that could be worthy

Reply:
If the current order is `PyQt5, PyQt6, PySide2, PySide6`, then I agree it shouldn't be changed on this PR.

Sorry for the noise guys, ignore my comments.